RevenueCat/Scripts/PresentedOfferingContext.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using RevenueCat.SimpleJSON;
+
+public partial class Purchases
+{
+    /// <summary>
+    /// Stores context about the offering a package was presented from.
+    /// </summary>
+    public class PresentedOfferingContext
+    {
+        public readonly string OfferingIdentifier;
+        [CanBeNull] public readonly string PlacementIdentifier;
+        [CanBeNull] public readonly PresentedOfferingTargetingContext TargetingContext;
+
+        public PresentedOfferingContext(JSONNode response)
+        {
+            OfferingIdentifier = response["offeringIdentifier"];
+            PlacementIdentifier = response["placementIdentifier"];
+            TargetingContext = new PresentedOfferingTargetingContext(response["targetingContext"]);
+        }
+
+        public override string ToString()
+        {
+            return $"{nameof(OfferingIdentifier)}: {OfferingIdentifier}\n" +
+                   $"{nameof(PlacementIdentifier)}: {PlacementIdentifier}\n" + 
+                   $"{nameof(TargetingContext)}: {TargetingContext}";
+        }
+
+        public string ToJson()
+        {
+            var contextDict = new JSONObject();
+            contextDict["offeringIdentifier"] = OfferingIdentifier;
+            if (PlacementIdentifier != null) {
+                contextDict["placementIdentifier"] = PlacementIdentifier;
+            }
+
+            if (TargetingContext != null) {
+                var targetingDict = new JSONObject();
```

Comment:
Would it be possible to use the `PresentedOfferingTargetingContext` type here? Not sure if we still have that problem with unity not serializing correctly if we do it like that though, so maybe this is ok.

Reply:
Ummm... it _might_ since it doesn't have any nullable fields 🤔 I'll give it a try!